tests/widgetresize.js

```diff
@@ -0,0 +1,324 @@
+
+	const commitStub = sinon.stub();
+	const mouseMock = {
+		down( editor, domTarget ) {
+			this._getPlugin( editor )._mouseDownListener( {}, {
+				target: domTarget
+			} );
+		},
+		move( editor, domTarget, eventData ) {
+			const combinedEventData = Object.assign( {}, eventData, {
+				target: domTarget
+			} );
+
+			this._getPlugin( editor )._mouseMoveListener( {}, combinedEventData );
+		},
+		up() {
+			this._getPlugin( editor )._mouseUpListener();
+		},
+
+		_getPlugin( editor ) {
+			return editor.plugins.get( WidgetResize );
+		}
+	};
+
+	before( () => {
+		mouseListenerSpies = {
+			down: sinon.spy( WidgetResize.prototype, '_mouseDownListener' ),
+			move: sinon.spy( WidgetResize.prototype, '_mouseMoveListener' ),
+			up: sinon.spy( WidgetResize.prototype, '_mouseUpListener' )
+		};
+	} );
+
+	after( () => {
+		for ( const stub of Object.values( mouseListenerSpies ) ) {
+			stub.restore();
+		}
+	} );
+
+	beforeEach( async () => {
+		editorElement = createEditorElement();
+		editor = await createEditor( editorElement );
+
+		setModelData( editor.model, '[<widget></widget>]' );
+
+		focusEditor( editor );
+
+		widget = editor.editing.view.document.getRoot().getChild( 0 );
+
+		for ( const stub of Object.values( mouseListenerSpies ) ) {
+			stub.resetHistory();
+		}
+		commitStub.resetHistory();
+
+		// It's crucial to have a precisely defined editor size for this test suite.
+		editor.editing.view.change( writer => {
+			const viewEditableRoot = editor.editing.view.document.getRoot();
+			writer.setAttribute( 'style', 'width: 400px; padding: 0px; overflow: hidden', viewEditableRoot );
+		} );
+	} );
+
+	afterEach( () => {
+		editorElement.remove();
+
+		if ( editor ) {
+			return editor.destroy();
+		}
+	} );
+
+	describe( 'plugin', () => {
+		it( 'is loaded', () => {
+			expect( editor.plugins.get( WidgetResize ) ).to.be.instanceOf( WidgetResize );
+		} );
+	} );
+
+	describe( 'mouse listeners', () => {
+		beforeEach( () => {
+			createResizer();
+		} );
+
+		it( 'doesnt break when called with unexpected element', async () => {
+			const unrelatedElement = document.createElement( 'div' );
+
+			editor.plugins.get( WidgetResize )._mouseDownListener( {}, {
+				target: unrelatedElement
+			} );
+		} );
+
+		it( 'passes new width to the options.onCommit()', async () => {
+			const usedResizer = 'top-right';
+			const domParts = getWidgetDomParts( widget, usedResizer );
+			const initialPointerPosition = getElementCenterPoint( domParts.widget, usedResizer );
+			const finalPointerPosition = Object.assign( {}, initialPointerPosition );
+
+			finalPointerPosition.pageX += 20;
+
+			mouseMock.down( editor, domParts.resizeHandle );
+
+			mouseMock.move( editor, domParts.resizeHandle, finalPointerPosition );
+			mouseMock.up();
+
+			expect( commitStub.callCount ).to.be.equal( 1 );
+			sinon.assert.calledWithExactly( commitStub, '120px' );
+		} );
+	} );
+
+	it( 'are detached when plugin is destroyed', async () => {
```

Comment:
Shouldn't this listener be in the previous describe (for mouse listeners)?

Reply:
Yes, this and the following one should have been moved to the `mouse listeners` suite.